test/tests/functional/pbs_multi_sched.py

```diff
@@ -0,0 +1,573 @@
+        j = Job(TEST_USER1, attrs={'Resource_List.select': '1:ncpus=1'})
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+        j = Job(TEST_USER1, attrs={'Resource_List.select': '1:ncpus=2'})
+        jid3 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid3)
+        self.server.delete(jid1, wait=True)
+        self.server.expect(JOB, 'job_state', op=UNSET, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+        j = Job(TEST_USER1, attrs={ATTR_queue: 'wq3',
+                                   'Resource_List.select': '1:ncpus=2'})
+        jid4 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
+        j = Job(TEST_USER1, attrs={ATTR_queue: 'wq3',
+                                   'Resource_List.select': '1:ncpus=1'})
+        jid5 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid5)
+        j = Job(TEST_USER1, attrs={ATTR_queue: 'wq3',
+                                   'Resource_List.select': '1:ncpus=2'})
+        jid6 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid6)
+        self.server.delete(jid4, wait=True)
+        self.server.expect(JOB, 'job_state', op=UNSET, id=jid4)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid6)
+
+    def test_run_limts_per_scheduler(self):
+        """
+        Test run_limits applied at server level is
+        applied for every scheduler seperately.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'max_run': '[u:PBS_GENERIC=1]'})
+        j = Job(TEST_USER1, attrs={'Resource_List.select': '1:ncpus=1'})
+        jid1 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        j = Job(TEST_USER1, attrs={'Resource_List.select': '1:ncpus=1'})
+        jid2 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid2)
+        j = Job(TEST_USER1, attrs={'Resource_List.select': '1:ncpus=1'})
+        jc = "Not Running: User has reached server running job limit."
+        self.server.expect(JOB, {'comment': jc}, id=jid2)
+        j = Job(TEST_USER1, attrs={ATTR_queue: 'wq3',
+                                   'Resource_List.select': '1:ncpus=1'})
+        jid3 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+        j = Job(TEST_USER1, attrs={ATTR_queue: 'wq3',
+                                   'Resource_List.select': '1:ncpus=1'})
+        jid4 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid4)
+        jc = "Not Running: User has reached server running job limit."
+        self.server.expect(JOB, {'comment': jc}, id=jid4)
+
+    def tearDown(self):
+        self.du.run_cmd(self.server.hostname, [
+                        'pkill', '-9', 'pbs_sched'], sudo=True)
+        TestFunctional.tearDown(self)
```

Comment:
@visheshh ,
I feel better place to do this sort of cleanup is "revert_to_defaults" method in sched class.
Also, does our code removes directories like: "/var/spool/pbs/sched_priv_sc1" which you are creating in your setup?
If no please make sure you clean/delete such dir and files.